tcms/static/js/utils.js

```diff
@@ -1,17 +1,16 @@
 /*
     Used to update a select when something else changes.
 */
-function updateSelect(data, selector, id_attr, value_attr) {
+function updateSelect(data, selector, id_attr, value_attr, group_attr) {
     var _select_tag = $(selector)[0];
     var new_options = '';
 
-    // in some cases, e.g. TestRun search, the 1st <option> element is ---
-    // which must always be there to indicate nothing selected
-    if (_select_tag.options.length) {
-        new_options = _select_tag.options[0].outerHTML;
-    }
-
+	currentGroup = '';
     data.forEach(function(element) {
+		if (currentGroup !== element[group_attr] && group_attr != null) {
```

Comment:
IMO the check for `null` should be first. You first check if an attribute isn't null then try to use it. 

Moreover you shouldn't check `group_attr !== null` but instead check whether such key exists in `element`.

Reply:
The intention behind a null value for `group_attr` was that if the calling function does not want `optgroup` based grouping, then it can set this parameter to null; otherwise it should provide the attribute it expects to contain the `optgroup` value for each element in data (to abstract specifics of how the HTML should look away from `updateSelect`).
Should I be using a different approach? If not, I suppose I should put a comment about this in the code

Reply:
I see what you mean but I still can't reconcile with the fact that we've got 2 conditions controlling the same thing. Let's revisit after the code has been simplified a bit after my newest comments.